File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.java
class Solution {
    public int[] topKFrequent(int[] nums, int k) {
        HashMap<Integer, Integer> countersPerNumber = new HashMap<>();

        for (int i = 0; i < nums.length; i++) {
            if (countersPerNumber.containsKey(nums[i])) {
                countersPerNumber.put(nums[i], countersPerNumber.get(nums[i]) + 1);
            } else {
                countersPerNumber.put(nums[i], 1);
            }
        }

        ArrayList<Map.Entry<Integer, Integer>> res = 
            new ArrayList(countersPerNumber.entrySet());
        res.sort(Map.Entry.comparingByValue());
        Collections.reverse(res);

        int[] result = new int[k];

        for (int i = 0; i < k; i++) {
            result[i] = res.get(i).getKey();
        }

        return result;
    }
}

File: Data Structures & Algorithms/subsets-ii/submission-1.py
class Solution:
    def subsetsWithDup(self, nums: List[int]) -> List[List[int]]:
        result = []

        nums.sort()

        current = []

        def dfs(i):
            if i >= len(nums):
                result.append(current.copy())
                return

            current.append(nums[i])
            dfs(i + 1)

            current.pop()
            while i < len(nums) - 1 and nums[i + 1] == nums[i]:
                i = i + 1
            dfs(i + 1)

        dfs(0)

        return result